### PR TITLE
BI-13522: upgrade consumer-maven-archetype to Java 21

### PR DESCRIPTION
**Pull request by tombradley1**

### CODE REVIEW

File: src/main/resources/archetype-resources/src/main/java/config/Config.java [hunk 1]
```diff
@@ -15,12 +16,23 @@
...
 
 @Configuration
 @EnableKafka
 public class Config {
 
+    @Bean
+    public ConcurrentHashMap<ServiceResultStatus, ResponseEntityFactory> responseEntityFactoryMap() {
+        return new ConcurrentHashMap<>();
+    }
+
```
Comment (VaughanJackson123) [inline src/main/resources/archetype-resources/src/main/java/config/Config.java:35]: So this was definitely required in your environment Tom? 
My initial testing seemed to suggest I did not need it. 
Best to keep it though anyway. 👍

File: src/main/resources/archetype-resources/src/main/java/service/Consumer.java [hunk 2]
```diff
@@ -40,7 +43,7 @@ public Consumer(Service service, MessageFlags messageFlags) {
             retryTopicSuffix = "-${consumer.group_id}-retry",
             dltTopicSuffix = "-${consumer.group_id}-error",
             dltStrategy = DltStrategy.FAIL_ON_ERROR,
-            fixedDelayTopicStrategy = FixedDelayStrategy.SINGLE_TOPIC,
+            sameIntervalTopicReuseStrategy = SameIntervalTopicReuseStrategy.SINGLE_TOPIC,
```
Comment (VaughanJackson123) [inline src/main/resources/archetype-resources/src/main/java/service/Consumer.java:46]: What is the reason for the change of strategy?

File: src/main/resources/archetype-resources/src/test/java/InvalidMessageRouterTest.java [hunk 3]
```diff
@@ -13,6 +13,8 @@
 import org.junit.jupiter.api.extension.ExtendWith;
 import org.mockito.Mock;
 import org.mockito.junit.jupiter.MockitoExtension;
+import ${package}.service.InvalidMessageRouter;
+import ${package}.util.MessageFlags;
```
Comment (VaughanJackson123) [inline src/main/resources/archetype-resources/src/test/java/InvalidMessageRouterTest.java:17]: These imports don't look right?

**Review by VaughanJackson123 — APPROVED** [review]
The PR title should probably be something like `BI-13522: upgrade consumer-maven-archetype to Java 21`?

File: src/main/resources/archetype-resources/src/main/java/service/Consumer.java (hunk 2 above)
Comment (tombradley1) [reply src/main/resources/archetype-resources/src/main/java/service/Consumer.java:46]: https://docs.spring.io/spring-kafka/docs/3.0.10/api/deprecated-list.html as seen here, FixedDelayStrategy is depreciated in the newer versions of spring-kafka.

File: src/main/resources/archetype-resources/src/test/java/ConsumerInvalidTopicTest.java [hunk 4]
```diff
@@ -46,8 +46,7 @@ void testPublishToInvalidMessageTopicIfInvalidDataDeserialised() throws Interrup
         if (!latch.await(5L, TimeUnit.SECONDS)) {
             fail("Timed out waiting for latch");
         }
-
-        ConsumerRecords<?, ?> consumerRecords = KafkaTestUtils.getRecords(testConsumer, 10000L, 2);
+        ConsumerRecords<?, ?> consumerRecords = KafkaTestUtils.getRecords(testConsumer, Duration.ofSeconds(1), 2);
```
Comment (VaughanJackson123) [inline src/main/resources/archetype-resources/src/test/java/ConsumerInvalidTopicTest.java]: `Duration.ofSeconds(1)` - was 10s before. Is 1 sec going to be enough?

File: src/main/resources/archetype-resources/src/main/java/config/Config.java (hunk 1 above)
Comment (tombradley1) [reply src/main/resources/archetype-resources/src/main/java/config/Config.java:35]: I found that without it, the tests all fail for me on a clean install.
